app/core/store/modules/classrooms.js

```diff
@@ -448,12 +448,35 @@ export default {
         commit('setMostRecentClassCode', codeCamel)
       }
     },
+
     fetchCourseLevels: async ({ commit, getters }, { classroomID, courseID }) => {
       if (getters.getCourseLevels(classroomID, courseID)) {
         return
       }
       const levels = await classroomsApi.getCourseLevels({ classroomID, courseID }, { data: { cacheEdge: true } })
       commit('setClassroomCourseLevels', { classroomID, courseID, levels })
-    }
+    },
+
+    addOrUpdateCourse: async ({ commit, getters}, { classroomId, courseId }) => {
+      const updatedClassroom = await classroomsApi.addOrUpdateCourse({ classroomId, courseId }, {})
+      commit('updateClassroomById', {
+        classroomID: classroomId,
+        updates: {
+          courses: updatedClassroom.courses
+        }
+      })
+      // Also update it in the classrooms by teacher, for the teacher dashboard
+      // TODO: this doesn't entirely work, takes a page refresh for the student to show up in the new course when switching to it--why?
```

Comment:
we're update a nested object which do not trigger vue's update by default

Reply:
i think change `Vue.set(state.classrooms.byTeacher, teacherId, teacherClassroomsState)` to `Vue.set(state.classrooms, 'ByTeacher', {..byTeacher object})` may help. 
but looks like we use prev vue.set in whole repo. maybe need more check